pandas/io/parsers.py

```diff
@@ -727,14 +706,80 @@ def parser_f(filepath_or_buffer,
 
 
 read_csv = _make_parser_function('read_csv', default_sep=',')
-read_csv = Appender(_read_csv_doc)(read_csv)
+read_csv = Appender(_parser_params.format(
+                    func_name='read_csv',
+                    summary=('Read a comma-separated values (csv) file '
+                             'into DataFrame.'),
+                    sep_doc=_sep_doc.format(default="','"),
+                    engine_doc=_engine_doc)
+                    )(read_csv)
 
 read_table = _make_parser_function('read_table', default_sep='\t')
-read_table = Appender(_read_table_doc)(read_table)
+read_table = Appender(_parser_params.format(
+                      func_name='read_table',
+                      summary="""Read general delimited file into DataFrame.
+
+.. deprecated:: 0.24.0
+Use :func:`pandas.read_csv` instead, passing ``sep='\\t'`` if necessary.""",
+                      sep_doc=_sep_doc.format(default="\\t (tab-stop)"),
+                      engine_doc=_engine_doc)
```

Comment:
I think only these two functions `read_csv` and `read_table` use `_parser_params` now. So, if that's the case, and both have the same value for `_engine_doc`, I think it makes more sense to have the content directly inside `_parser_params`. And not have this variable anymore, right?

Other than that looks good to me. I think now the validation should be happy with the 3 functions, right?

Reply:
I also remove `_sep_doc` variable and merge it inside `_parser_params`. Let me know if you want the other way. Now the validation is fine with all these three 👍 .